notebooks/GridSearchCV for Neural Networks.ipynb

```diff
@@ -0,0 +1,541 @@
+    "\n",
+    "renamed_columns = [column.rstrip('score')+'error' for column in columns_to_rename]\n",
+    "grid_search_results[renamed_columns] = grid_search_results[columns_to_rename]\n",
+    "grid_search_results = grid_search_results.drop(columns_to_rename,axis=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, let's look at the basic descriptive statistics of the test and train error."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "columns_to_describe = ['mean_test_error','mean_train_error']\n",
+    "\n",
+    "grid_search_results[columns_to_describe].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The distribution appears to be very skew,\n",
+    "with the maximum being 11 orders of magnitude above the minimum\n",
+    "and 10 orders of magnitude larger than the 75th percentile.\n",
+    "\n",
+    "These bad performers will make it difficult to visualize\n",
+    "the remaining data, so we drop them\n",
+    "by `query`ing for models that have reasonable performance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "filtered_results = grid_search_results.query('mean_test_error<1')\n",
+    "filtered_results = filtered_results.query('mean_train_error<1')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can visualize our results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(12,6))\n",
+    "sns.distplot(filtered_results.mean_test_error, kde=False,\n",
+    "             hist_kws={'normed':True,\n",
+    "                       'edgecolor':'k',\n",
+    "                       'linewidth':4,\n",
+    "                      'histtype':'stepfilled'},\n",
+    "            label='test');\n",
+    "\n",
+    "sns.distplot(filtered_results.mean_train_error, kde=False,\n",
+    "             hist_kws={'normed':True,\n",
+    "                       'edgecolor':'k',\n",
+    "                       'linewidth':4,\n",
+    "                      'histtype':'stepfilled'},\n",
+    "            label='train');\n",
+    "plt.legend(fontsize=20)\n",
+    "plt.xlabel('MCRMS Error',fontsize=24);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Interestingly, it appears that there's a hard limit to our performance on the test set,\n",
+    "while performance on the training set can reach up to nearly perfect.\n",
+    "\n",
+    "Though it's difficult to compare precisely,\n",
+    "since the test set used in the Kaggle competition \n",
+    "is now unavailable,\n",
+    "this limit to our performance is close to\n",
```

Comment:
I think part of your thought got cut off here - but I assume you're talking about the limit in performance of the scores seen on the kaggle leaderboards?